### PR TITLE
Merge overwrite

### PR DESCRIPTION
**Pull request by dstuebe**
Fixed bug 499 to make overwrite work properly for hfradar data.

### CODE REVIEW

File: ion/services/dm/ingestion/ingestion.py [hunk 1]
```diff
@@ -1296,7 +1406,11 @@ class IngestionService(ServiceProcess):
         log.debug('>> sup     min_offset    = %i' % merge_dims[sup_agg_dim_name].min_offset)
         log.debug('>> sup_sindex (adjusted) = %i' % sup_sindex)
         log.debug('>> sup_eindex (adjusted) = %i' % sup_eindex)
+        '''
+
+
         for merge_var in sup_root.variables:
+            yield pu.asleep(0.001)
```
Comment (daf) [inline ion/services/dm/ingestion/ingestion.py]: I assume this is just to keep the reactor from busylooping?

File: ion/services/dm/ingestion/ingestion.py [hunk 2]
```diff
@@ -1876,35 +1985,66 @@ class IngestionService(ServiceProcess):
...
+
         # Make sure the actual length of the new ndarray matches the length indicated by
         # the cumulative sizes of the arrays bounds
-        expect_len = reduce(lambda x,y: x+y, [bound.size for bound in new_ba.bounds])
+
         actual_len = len(new_nd.value)
         assert(expect_len == actual_len, 'Actual length of ndarray != expected length: %i != %i' % (actual_len, expect_len))
 
-
+        log.info('Alomst DOne!!!')
```
Comment (daf) [inline ion/services/dm/ingestion/ingestion.py]: Should these remain in here as info level?

File: ion/services/dm/ingestion/ingestion.py [hunk 3]
```diff
@@ -1105,7 +1109,25 @@ class IngestionService(ServiceProcess):
...
+
+
+            # Get the agg vars - if this is not an FMRC...
+            if cur_agg_var is None:
+                try:
+                    cur_agg_var = cur_root.FindVariableByName(sup_agg_dim_name)
+                except OOIObjectError, oe:
+                    log.exception('Time Variable name does not match its dimension name')
+                    raise IngestionError('Can not get current dataset Time Variable: "%s", in dataset: %s' % (sup_agg_dim_name, self.dataset.repository_key) )
+
```
Comment (daf) [inline ion/services/dm/ingestion/ingestion.py:1121]: These if blocks are safe to run here?  they used to exist under the block  `if sup_fcst_dim_name is not None and not len(sup_fcst_dim_name) == 0:`

File: ion/services/dm/ingestion/ingestion.py (hunk 1 above)
Comment (dstuebe) [inline ion/services/dm/ingestion/ingestion.py]: yes

File: ion/services/dm/ingestion/ingestion.py (hunk 2 above)
Comment (dstuebe) [inline ion/services/dm/ingestion/ingestion.py]: No :-(

File: ion/services/dm/ingestion/ingestion.py (hunk 3 above)
Comment (dstuebe) [inline ion/services/dm/ingestion/ingestion.py:1121]: Yes - that is specific to the FMRC ingestion method - other time variables are still pulled there. This is general for any path through this (mess-of-a) method.

File: ion/services/dm/ingestion/ingestion.py (hunk 1 above)
Comment (dstuebe) [inline ion/services/dm/ingestion/ingestion.py]: Removed it...

**Comment by daf** [comment]
[Reviewed]

**Pull request merged**